dockstore-webservice/src/main/java/io/dockstore/webservice/core/Collection.java

```diff
@@ -82,10 +82,10 @@
     private long id;
 
     @Column(nullable = false)
-    @Pattern(regexp = "[a-zA-Z][a-zA-Z\\d]*")
+    @Pattern(regexp = "[a-zA-Z](-?[a-zA-Z\\d]){0,38}")
```

Comment:
My regex is weak, but shouldn't it be `{2,38}` to match the `@Size` on the next line?

Reply:
Kinda seems like it should be, but if you make the minimum 2, it can't generate `a-b`, which is legal according to our description.  And, if you make it 1, it's gonna generate some "short" strings.  Fortunately, like the current regexp, the adjacent `@Size` annotation ensures a minimum length of 3.  So, might as well go with 0, it's as easy to understand and as correct as anything else. :)